lib,test,tools: rename blacklist/whitelist to denylist/allowlist

mkrawczuk opened the pull request:
In contrary to groundless and silly whitelist/blacklist,
allowlist/blocklist is concise and self-explanatory.

This commit intentionally does not change minified scripts, deps,
node_modules and changelogs.
Occurences in dependencies and node_modules should be replaced in
adequate repositories of the projects Node depends upon.
Occurences in changelogs were left as-is because changing them would
bring confusion with related issue titles, commit messages, etc.

<!--
Thank you for your pull request. Please provide a description above and review
the requirements below.

Bug fixes and new features should include tests and possibly benchmarks.

Contributors guide: https://github.com/nodejs/node/blob/master/CONTRIBUTING.md
-->

##### Checklist
<!-- Remove items that do not apply. For completed items, change [ ] to [x]. -->

- [x] `make -j4 test` (UNIX), or `vcbuild test` (Windows) passes
- [x] commit message follows [commit guidelines](https://github.com/nodejs/node/blob/master/doc/guides/contributing/pull-requests.md#commit-message-guidelines)

<!--
Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
-->

mkrawczuk commented:
For the record: I have used this little bash script to avoid changing the occurences manually:
```
OCCURENCES=$(grep -rnil \
--exclude="CHANGELOG*" \
--exclude="*.min.js" \
--exclude-dir=".git" \
--exclude-dir="deps" \
--exclude-dir="node_modules" \
"blacklist" .)
echo $OCCURENCES | xargs -l sed -i -e "s/blacklist/blocklist/g"
echo $OCCURENCES | xargs -l sed -i -e "s/Blacklist/Blocklist/g"
echo $OCCURENCES | xargs -l sed -i -e "s/BLACKLIST/BLOCKLIST/g"

OCCURENCES=$(grep -rnil \
--exclude="CHANGELOG*" \
--exclude="*.min.js" \
--exclude-dir=".git" \
--exclude-dir="deps" \
--exclude-dir="node_modules" \
"whitelist" .)
echo $OCCURENCES | xargs -l sed -i -e "s/a whitelist/an allowlist/g"
echo $OCCURENCES | xargs -l sed -i -e "s/whitelist/allowlist/g"
echo $OCCURENCES | xargs -l sed -i -e "s/Whitelist/Allowlist/g"
echo $OCCURENCES | xargs -l sed -i -e "s/WHITELIST/ALLOWLIST/g"
```

richardlau commented:
> 
> 
> I think the files in `tools/gyp` are maintained elsewhere and just copied into this repo, but I don’t know where we pull them in from. @nodejs/gyp

https://github.com/nodejs/gyp-next

Trott commented:
`tools/lint-md.js` is generated and should not be edited. The term comes from the upstream dependency `remark-lint-maximum-line-length` and not in our code, so the changes will have to be made there. Otherwise the term will just re-appear the next time `lint-md.js` is built.

Trott commented:
+1 on the change generally. I'd prefer `allow-list` and `block-list`. Probably will want to make sure verb instances are replaced with something else too. Those are my preferences, but I'm not blocking this.

Trott commented on lib/internal/bootstrap/loaders.js:
```diff
@@ -64,11 +64,11 @@ ObjectDefineProperty(process, 'moduleLoadList', {
-// internalBindingWhitelist contains the name of internalBinding modules
-// that are whitelisted for access via process.binding()... This is used
+// internalBindingAllowlist contains the name of internalBinding modules
+// that are allowlisted for access via process.binding()... This is used
```
```suggestion
// that are allowed for access via process.binding()... This is used
```

Trott commented on test/parallel/test-process-binding-internalbinding-allowlist.js:
```diff
@@ -4,7 +4,7 @@
 const assert = require('assert');
 
-// Assert that whitelisted internalBinding modules are accessible via
+// Assert that allowlisted internalBinding modules are accessible via
```
```suggestion
// Assert that allowed internalBinding modules are accessible via
```

Trott commented on tools/gyp/pylib/gyp/generator/ninja.py:
```diff
@@ -79,7 +79,7 @@ def QuoteShellArgument(arg, flavor):
   by the shell."""
   # Rather than attempting to enumerate the bad shell characters, just
-  # whitelist common OK ones and quote anything else.
+  # allowlist common OK ones and quote anything else.
```
```suggestion
  # allow common OK ones and quote anything else.
```

Trott commented on tools/lint-md.js:
```diff
@@ -46416,29 +46416,29 @@ function maximumLineLength(tree, file, option) {
   }
 
-  // Finally, whitelist some inline spans, but only if they occur at or after
+  // Finally, allowlist some inline spans, but only if they occur at or after
```
```suggestion
  // Finally, allow some inline spans, but only if they occur at or after
```

Trott commented on tools/lint-md.js:
```diff
@@ -46416,29 +46416,29 @@ function maximumLineLength(tree, file, option) {
   // the wrap.
   // However, when they do, and there’s whitespace after it, they are not
-  // whitelisted.
+  // allowlisted.
```
```suggestion
  // allowed.
```

Trott commented on tools/lint-md.js:
```diff
@@ -46416,29 +46416,29 @@ function maximumLineLength(tree, file, option) {
     var final;
 
-    /* istanbul ignore if - Nothing to whitelist when generated. */
+    /* istanbul ignore if - Nothing to allowlist when generated. */
```
```suggestion
    /* istanbul ignore if - Nothing to allow when generated. */
```

Trott commented on tools/lint-md.js:
```diff
@@ -46416,29 +46416,29 @@ function maximumLineLength(tree, file, option) {
     final = end$8(node);
 
-    // No whitelisting when starting after the border, or ending before it.
+    // No allowlisting when starting after the border, or ending before it.
```
```suggestion
    // Nothing to allow when starting after the border, or ending before it.
```

DerekNonGeneric replied:
Same as below.

Trott commented on tools/lint-md.js:
```diff
@@ -46416,29 +46416,29 @@ function maximumLineLength(tree, file, option) {
     }
 
-    // No whitelisting when there’s whitespace after the link.
+    // No allowlisting when there’s whitespace after the link.
```
```suggestion
    // Nothing to allow when there’s whitespace after the link.
```

DerekNonGeneric replied:
Suggestion: `Do not allow`

Trott commented on tools/lint-md.js:
```diff
@@ -46447,18 +46447,18 @@ function maximumLineLength(tree, file, option) {
 
-  // Whitelist from `initial` to `final`, zero-based.
-  function whitelist(initial, final) {
+  // Allowlist from `initial` to `final`, zero-based.
```
```suggestion
  // Allow from `initial` to `final`, zero-based.
```

Trott commented on configure.py:
```diff
@@ -838,7 +838,7 @@ def get_gas_version(cc):
 # Note: Apple clang self-reports as clang 4.2.0 and gcc 4.2.1.  It passes
 # the version check more by accident than anything else but a more rigorous
-# check involves checking the build number against a whitelist.  I'm not
+# check involves checking the build number against an allowlist.  I'm not
```
```suggestion
# check involves checking the build number against an allow-list.  I'm not
```

Trott commented on test/common/README.md:
```diff
@@ -592,7 +592,7 @@ If set, crypto tests are skipped.
 
 A comma-separated list of variables names that are appended to the global
-variable whitelist. Alternatively, if `NODE_TEST_KNOWN_GLOBALS` is set to `'0'`,
+variable allowlist. Alternatively, if `NODE_TEST_KNOWN_GLOBALS` is set to `'0'`,
```
```suggestion
variable allow-list. Alternatively, if `NODE_TEST_KNOWN_GLOBALS` is set to `'0'`,
```

Trott commented on tools/cpplint.py:
```diff
@@ -4119,8 +4119,8 @@ def CheckTrailingSemicolon(filename, clean_lines, linenum, error):
   # brace initialization, there are more places where semicolons are
-  # required than not, so we use a whitelist approach to check these
-  # rather than a blacklist.  These are the places where "};" should
+  # required than not, so we use an allowlist approach to check these
```
```suggestion
  # required than not, so we use an allow-list approach to check these
```

Trott commented on tools/cpplint.py:
```diff
@@ -4119,8 +4119,8 @@ def CheckTrailingSemicolon(filename, clean_lines, linenum, error):
-  # required than not, so we use a whitelist approach to check these
-  # rather than a blacklist.  These are the places where "};" should
+  # required than not, so we use an allowlist approach to check these
+  # rather than a blocklist.  These are the places where "};" should
```
```suggestion
  # rather than a block-list.  These are the places where "};" should
```

DerekNonGeneric replied:
The opposite of allow is deny.

Trott commented on tools/cpplint.py:
```diff
@@ -4177,11 +4177,11 @@ def CheckTrailingSemicolon(filename, clean_lines, linenum, error):
     #  - EXCLUSIVE_LOCKS_REQUIRED, SHARED_LOCKS_REQUIRED, LOCKS_EXCLUDED:
     #
-    # We implement a whitelist of safe macros instead of a blacklist of
+    # We implement an allowlist of safe macros instead of a blocklist of
```
```suggestion
    # We implement an allow-list of safe macros instead of a block-list of
```

Trott commented on tools/cpplint.py:
```diff
@@ -4177,11 +4177,11 @@ def CheckTrailingSemicolon(filename, clean_lines, linenum, error):
     # google code and would have been easier to implement.  This is because
-    # the downside for getting the whitelist wrong means some extra
-    # semicolons, while the downside for getting the blacklist wrong
+    # the downside for getting the allowlist wrong means some extra
```
```suggestion
    # the downside for getting the allow-list wrong means some extra
```

Trott commented on tools/cpplint.py:
```diff
@@ -4177,11 +4177,11 @@ def CheckTrailingSemicolon(filename, clean_lines, linenum, error):
-    # the downside for getting the whitelist wrong means some extra
-    # semicolons, while the downside for getting the blacklist wrong
+    # the downside for getting the allowlist wrong means some extra
+    # semicolons, while the downside for getting the blocklist wrong
```
```suggestion
    # semicolons, while the downside for getting the block-list wrong
```

DerekNonGeneric replied:
Same as above.

Trott commented on tools/cpplint.py:
```diff
@@ -5468,19 +5468,19 @@ def CheckForNonConstReference(filename, clean_lines, linenum,
     return
   elif not Search(r'\S+\([^)]*$', line):
-    # Don't see a whitelisted function on this line.  Actually we
+    # Don't see an allowlisted function on this line.  Actually we
```
```suggestion
    # Don't see an allowed function on this line.  Actually we
```

DerekNonGeneric commented on lib/internal/bootstrap/loaders.js:
```diff
@@ -64,11 +64,11 @@ ObjectDefineProperty(process, 'moduleLoadList', {
 // to provide a transition path for modules that are being moved over to
 // internalBinding.
-const internalBindingWhitelist = new SafeSet([
+const internalBindingAllowlist = new SafeSet([
```
```suggestion
const internalBindingAllowList = new SafeSet([
```

DerekNonGeneric commented on tools/gyp/PRESUBMIT.py:
```diff
@@ -10,7 +10,7 @@
 
 
-PYLINT_BLACKLIST = [
+PYLINT_BLOCKLIST = [
```
```suggestion
PYLINT_DENYLIST = [
```

DerekNonGeneric commented on tools/gyp/PRESUBMIT.py:
```diff
@@ -111,14 +111,14 @@ def CheckChangeOnCommit(input_api, output_api):
   try:
     sys.path = ['pylib', 'test/lib'] + sys.path
-    blacklist = PYLINT_BLACKLIST
+    blocklist = PYLINT_BLOCKLIST
```
```suggestion
    blocklist = PYLINT_BLOCK_LIST
```

Pulfer commented:
Please stop this madness. whitelist/blacklist is common. If Google does something stupid it doesn't mean everyone else should do the same.

mkrawczuk commented:
A couple of fixups:
changes made to `tools/gyp` have been reverted, except for the file `PRESUBMIT.py`, because I could not find it in the original repo. Relevant PR:
https://github.com/nodejs/gyp-next/pull/46

Reverted changes made to `tools/lint-md.js`, since it is an external dependency. Maybe will open a PR there.

Occurences of 'allowlisted' have been changed to a more concise 'allowed'.

According to @DerekNonGeneric's suggestion, changed `blocklist` to `denylist` since "deny" is opposite to "allow". Also changed this PR's title.

I decided not to go along with @Trott's sugestion to change `allowlist` to `allow-list` and `denylist` (previously: `blocklist`) to `deny-list` (prevously: `block-list`) since `-` is one of the forbidden signs for variable names in JS. There is risk that it might bring confusion in the future.

aduh95 commented:
@mkrawczuk this needs a rebase to resolve the git conflicts.

Trott commented:
I did a checkout, squash, rebase, resolve conflicts, and there were no changes left. Sorry we didn't land this sooner and thanks for helping us improve language used in the codebase.
